File: apps/desktop/src-tauri/src/pty/commands.rs
use super::terminal::TERMINAL_MANAGER;
use tauri::AppHandle;

#[tauri::command]
pub fn spawn_terminal(
    app_handle: AppHandle,
    cols: u16,
    rows: u16,
    cwd: Option<String>,
) -> Result<String, String> {
    let mut manager = TERMINAL_MANAGER.lock();
    manager.create(app_handle, cols, rows, cwd)
}

#[tauri::command]
pub fn write_terminal(id: String, data: String) -> Result<(), String> {
    let mut manager = TERMINAL_MANAGER.lock();
    manager.write(&id, &data)
}

#[tauri::command]
pub fn resize_terminal(id: String, cols: u16, rows: u16) -> Result<(), String> {
    let manager = TERMINAL_MANAGER.lock();
    manager.resize(&id, cols, rows)
}

#[tauri::command]
pub fn kill_terminal(id: String) -> Result<(), String> {
    let mut manager = TERMINAL_MANAGER.lock();
    manager.kill(&id)
}

File: apps/desktop/src/lib/ipc/commands.ts
import { invoke } from "@tauri-apps/api/core";
import { listen, type UnlistenFn } from "@tauri-apps/api/event";

// File system types
export interface FileEntry {
  name: string;
  path: string;
  is_dir: boolean;
  is_file: boolean;
  is_hidden: boolean;
  size?: number;
  modified?: number;
}

export interface ProjectInfo {
  name: string;
  path: string;
  git_branch?: string;
  git_status?: string;
}

// Terminal commands
export async function spawnTerminal(
  cols: number,
  rows: number,
  cwd?: string,
): Promise<string> {
  return invoke<string>("spawn_terminal", { cols, rows, cwd });
}

export async function writeTerminal(id: string, data: string): Promise<void> {
  return invoke<void>("write_terminal", { id, data });
}

export async function resizeTerminal(
  id: string,
  cols: number,
  rows: number,
): Promise<void> {
  return invoke<void>("resize_terminal", { id, cols, rows });
}

export async function killTerminal(id: string): Promise<void> {
  return invoke<void>("kill_terminal", { id });
}

// Terminal output listener
export function onTerminalOutput(
  terminalId: string,
  callback: (data: string) => void,
): Promise<UnlistenFn> {
  return listen<string>(`pty-output-${terminalId}`, (event) => {
    callback(event.payload);
  });
}

// File system commands
export async function readDirectory(path: string): Promise<FileEntry[]> {
  return invoke<FileEntry[]>("read_directory", { path });
}

export async function getProjectInfo(path: string): Promise<ProjectInfo> {
  return invoke<ProjectInfo>("get_project_info", { path });
}

export async function readFileContent(path: string): Promise<string> {
  return invoke<string>("read_file_content", { path });
}
